doc/subsystems/shell/shell.rst

```diff
@@ -435,4 +435,21 @@ the shell will only print the subcommands registered for this command:
 
 	  params  ping
 
+Shell as the logger backend
+***************************
+
+Shell instance can act as the :ref:`logger` backend. Shell ensures that log
+messages are correctly multiplexed with shell output. Log messages from logger
+thread are enqueued and processed in the shell thread. Logger thread will block
```

Comment:
What if logs are processed from user thread or by direct calls to LOG_PROCESS()?

Reply:
this context still can be called logger thread context. Do you consider case when LOG_PROCESS() is called from context that cannot be blocked? If so then I can add assert to that call to ensure that message is passed from thread context.